cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -651,11 +652,10 @@ func main() {
 		}()
 
 		err := proxyClient.Shutdown(*termTimeout)
-		if err == nil {
-			os.Exit(0)
+		if err != nil {
+			logging.Errorf("Error during SIGTERM shutdown: %v", err)
+			os.Exit(1)
```

Comment:
This os.Exit here - does this still ensure `cancel()` is called?

Reply:
I double checked and interestingly, no. A call to `os.Exit` exits immediately ignoring any defers.

Cf. https://stackoverflow.com/questions/27629380/how-to-exit-a-go-program-honoring-deferred-calls